src/util/features.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from 'react';
+import get from 'lodash/get';
+
+export type FeatureOptions = {
+    enabled: Boolean,
+    [key: string]: any,
+};
+
+export type FeatureConfig = {
+    [key: string]: FeatureOptions,
+};
+
+function isFeatureEnabled(features: FeatureConfig, featureName: string) {
+    return !!get(features, `${featureName}.enabled`, false);
+}
+
+function getFeatureConfig(features: FeatureConfig, featureName: string) {
+    return get(features, `${featureName}`, {});
+}
+
+const FeatureContext = React.createContext({});
+
+function FeatureProvider({ features = {}, children }: { features?: FeatureConfig, children: ?React.Node }) {
+    return <FeatureContext.Provider value={features}>{children}</FeatureContext.Provider>;
+}
+
+const FeatureConsumer = FeatureContext.Consumer;
+
+function FeatureFlag({
+    feature,
+    enabled = () => null,
+    disabled = () => null,
+    children,
+}: {
+    feature: string,
+    enabled?: FeatureOptions => React.Node,
+    disabled?: FeatureOptions => React.Node,
+    children?: (Boolean, FeatureOptions) => React.Node,
+}) {
+    return (
+        <FeatureContext.Consumer>
+            {features => {
+                const isEnabled = isFeatureEnabled(features, feature);
+                const featureConfig = getFeatureConfig(features, feature);
+                if (children) return children(isEnabled, featureConfig);
```

Comment:
children should not be a function, should be a simple React.Node

Reply:
That removes the ability to provide a fallback element, but enabled/disabled could be used instead. OK.

Reply:
Also, currently the feature format allows additional parameters in the feature object. These are passed to the function as arguments. Should this be removed?

```js
const features = {
  aFeature: {
    enabled: true,
    fetchUrl: '/app-api/abc',
    isNameRequired: false,
  }
}
```

Reply:
Initially I was thinking roughly along the lines of this...

```jsx
// Somewhere in EUA
const features = {
    activityfeed: {
        hasNewUI: FF boolean from monolith,
    },
};

// withFeatureContext.js
const withFeatureContext = (WrappedComponent: React.ComponentType<any>, feature) => (props: any) => (
    <FeatureContext.Consumer>
        {features => <WrappedComponent {...props} {...features[feature]} />}
    </FeatureContext.Consumer>
);

// ActivityFeed.js
class ActivityFeed {
    render() {
        this.props.hasNewUI ? <NewUI> : <OldUI>
    }
}

export withFeatureContext(ActivityFeed, 'activityfeed')
```